### PR TITLE
[NEW API]: removeExisting to remove all fauxHandlers

### PR DESCRIPTION
**Pull request by snewcomer**
https://github.com/JarrodCTaylor/fauxjax/issues/26

### CODE REVIEW

**Comment by JarrodCTaylor** [comment]
I will give this a look over the weekend

**Comment by JarrodCTaylor** [comment]
I appreciate the pull request and the enthusiasm for continuing to evolve the library. However, I need a little more convincing as I have reservations about including this method in the API. Just looking at the names knowing nothing about the library I would be confused as to what the difference is between `remove` and `removeExisting`. The only things being checked for equality in order to remove a handler by this proposed method are `url` and `method`. Which means there are other possible permutations of things that could be different between two handlers that get ignored leaving the door open to introduce some subtle and annoying errors. 

I am also still unclear on what scenario this is providing a solution for? Why is the index of the handler not available from when it was created?

**Comment by TayHobbs** [comment]
I still fail to see the advantage to this over storing the index of this and using remove, or even `clear();`?

Agreed with @JarrodCTaylor too, if there is a reason for needing this, then it should definitely be checking more than just `url` and `method`. Especially if `strictMatching` is enabled.

**Comment by snewcomer** [comment]
The index definitely could be found, but now you have to manage that state explicitly rather than just dumping existing xhrs.  `clear` can be problematic, b/c let's say you have a view that is a `list`, the `detail` slides over top, and there is also fetching permissions data`.  So calling clear would require me to mock out list/detail/permissions again, even though I already set it up earlier in the test.  `removeExistingByUrlMethod`?

Here is what the functionality of this API allow me to do.  Let's say I visit an edit view (a `person`), change their `role` in that view that causes that detail route to re-fire w/o refreshing the page.  So in this case, I mocked out two person detail fetches, but those two have different payloads (ie. first fetch has data that you can edit that view, but second fetch, since you changed their role, you can't edit that role).  However, in this case, the first mocked out fetch wins, which is problematic.

So from the original issue (https://github.com/JarrodCTaylor/fauxjax/issues/26):

**Comment by JarrodCTaylor** [comment]
I still don't follow the issue with hanging onto the index value returned when the mock was created for use in removal. I'm going to assume that you have your own valid reasons within your code base. However, I continue to believe that those reasons are unique to your codebase and are not ubiquitous enough to justify the addition of a method designed specifically to address them.
 
As a compromise what do you think about adding a `$.fauxjax.fauxHandlers()` that just returns the `fauxHandlers` array? That way you could do any kind of manipulation you want based on whatever information you have or want to use, and it will not introduce any confusion into the API or allow the potential to accidentally create errors if used improperly as the other method possibly could.

**Comment by snewcomer** [comment]
So it isn't a terrible problem to have to manage the index.  BUT there is one basic problem with `fauxjax.new`.  It does not allow multiple xhrs in one test with the same url/method.  The first one always wins.  So, yes managing the index is one sol'n to a problem we all agree is present.  Whether I went down the route of using the index returned from `fauxjax.new` or getting `fauxHandlers`, it still results in the same overhead.  In a test, I need to assign the index to a variable (or something else), and call `remove` at the right time.  Doing that 100 times is annoying.  This pull request turns something that is repetitive to an abstraction to remove mocked xhrs.  I can write a wrapper called `ajax` and do something like this for example: 

```
function ajax(url, verb, data, headers, status, response) {
    Ember.$.fauxjax.removeExisting(url, verb);
    xhr(url, verb, data, headers, status, response);
}
```

and call

```
ajax( PEOPLE_URL, 'GET', null, {}, 200, some_data );
...do some things in the test
ajax( PEOPLE_URL, 'GET', null, {}, 200, new_data );
```

instead of 

```
const indx = xhr( PEOPLE_URL, 'GET', null, {}, 200, some_data );
...do some things in the test
fauxjax.remove(indx);
xhr( PEOPLE_URL, 'GET', null, {}, 200, new_data );
```

Is there anything wrong with `remove`?  I don't think so.  Can it be made better?  I would assume yes.

**Comment by JarrodCTaylor** [comment]
I believe you are using the library incorrectly. Fauxjax DOES properly process multiple handlers that have the same url/method and different data. UNLESS you have disabled [strictMatching](https://github.com/JarrodCTaylor/fauxjax/wiki/API-Reference#strict-matching). I have added a [test to demonstrate](https://github.com/JarrodCTaylor/fauxjax/blob/master/test/dataContents.coffee#L53)

So if that is your primary issue then I believe all of this is moot. 

Additionally your `ajax` wrapper could just as easily be written as:
```
function ajax(url, verb, data, headers, status, response, existingHandlerIndex) {
    if (existingHandlerIndex) { $.fauxjax.remove(existingHandlerIndex); }
    xhr(url, verb, data, headers, status, response);
}
```
 you would end up with the exact solution you are desiring using the existing API
